### PR TITLE
Update motivations references in Search 2.0

### PR DESCRIPTION
**Pull request by kirschbombe**
Preview: https://preview.iiif.io/api/search-v3-alignment-motivations/search/2.0/

Addresses part of issue #2053
- Removes the table of motivations in Search 2.0
- Replaces the table with a paragraph pointing readers to the motivations registry and mentioning the two motivations discussed in Search 2.0 - update: moves this text to the `motivation` row of the params table
- Adds links to the motivations registry for `contextualizing` and `highlighting` that point to their respective sections in Search 2.0
- Also merges `main` into the branch for the motivations registry updates
- Also, removes 4 references to `Layer`, and in one case replaces language with Annotation Collection (could use some feedback on whether I got this right) - #2126  
- Updates mentions of the API to consistently use "Content Search"

### CODE REVIEW

File: source/_includes/links.md [hunk 1]
```diff
@@ -18,9 +18,10 @@
...
-[contentstate-export]: {{ site.api_url | absolute_url }}/content-state/0.9/#export "3.7. Exporting Current Content State from Viewer" 
+[contentstate-export]: {{ site.api_url | absolute_url }}/content-state/0.9/#export "3.7. Exporting Current Content State from Viewer"
 [contentstate-http]: {{ site.api_url | absolute_url }}/content-state/0.9/#7-http-requests-and-responses "HTTP Request and Responses"
 [contentstate-initialization]: {{ site.api_url | absolute_url }}/content-state/0.9/#initialization "Initialization mechanisms (protocol)"
+[contentstate10-22-form-of-annotation]: {{ site.api_url | absolute_url }}/content-state/1.0/#22-form-of-annotation
 [btoa-unicode-problem]: https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem "The Unicode Problem"
 [contentstate-encoding]: {{ site.api_url | absolute_url }}/content-state/0.9/#6-content-state-encoding "Content State Section 6"
 [contentstate-export]: {{ site.api_url | absolute_url }}/content-state/0.9/#export
```
Comment (hadro) [inline source/_includes/links.md:27]: Totally minor, but looks like 'contentstate-export' appears twice in links.md, line 21 and line 27. It's not throwing errors, but I got a bunch of warnings when I looked at this briefly locally

Comment (kirschbombe) [reply source/_includes/links.md]: I removed the duplicate 'contentstate-export' in links.md.

**Review by digitaldogsbody — APPROVED** [review]

**Review by zimeon — APPROVED** [review]

File: source/_includes/links.md [hunk 2]
```diff
@@ -18,12 +18,12 @@
 [contentstate03]: {{ site.api_url | absolute_url }}/content-state/0.3/ "IIIF Content State API 0.3"
 [contentstate09]: {{ site.api_url | absolute_url }}/content-state/0.9/ "IIIF Content State API 0.9"
 [contenstate-stable-version]: {{ site.api_url | absolute_url }}/content-state/{{ site.data.apis.content-state.latest.major }}.{{ site.data.apis.content-state.latest.minor }}/ "Latest Content State API Stable Version"
-[contentstate-export]: {{ site.api_url | absolute_url }}/content-state/0.9/#export "3.7. Exporting Current Content State from Viewer" 
+[contentstate-export]: {{ site.api_url | absolute_url }}/content-state/0.9/#export "3.7. Exporting Current Content State from Viewer"
```
Comment (digitaldogsbody) [inline source/_includes/links.md:21]: Now that Content State is at 1.0, should this be updated?

File: source/search/2.0/index.md [hunk 3]
```diff
@@ -322,7 +308,7 @@ There may be properties that are specific to the search result, and not features
...
 
-To incrementally build upon existing solutions and provide graceful degradation for clients that do not support these features and retain compatibility with the Presentation API, the search API specific information is included in a second list of annotations called `annotations`, other than the `ignored` property. This structure mirrors the distinction in the [Presentation API][prezi3], where the main content annotations are listed in `items` and additional annotations such as comments are listed in `annotations`. 
+To incrementally build upon existing solutions and provide graceful degradation for clients that do not support these features and retain compatibility with the Presentation API, the search API specific information is included in a second list of annotations called `annotations`, other than the `ignored` property. This structure mirrors the distinction in the [Presentation API][prezi3], where the main content annotations are listed in `items` and additional annotations such as comments are listed in `annotations`.
```
Comment (digitaldogsbody) [inline source/search/2.0/index.md:311]: Should we standardise the name and capitalisation? Here is is "search API", but on line 309 it is "Search API", and in other places "Content Search API"

Comment (kirschbombe) [reply source/search/2.0/index.md]: I went through the doc and updated any references to the API to use "Content Search". Also updated case for Canvas, Manifest, and Range throughout.

**Review by digitaldogsbody — APPROVED** [review]

**Comment by kirschbombe** [comment]
Merged since everyone seems OK with the motivations updates.

**Pull request merged**